### PR TITLE
Add records `n` beats feature.

### PR DESCRIPTION
**Pull request by vale981**
This is a squashed version of my earlier pull request.
It allows to select the number of beats to record in advance and contains some tweaks to the context menu code.

### CODE REVIEW

**Comment by georgkrause** [comment]
Great thanks! Will do some testing tomorrow!

**Comment by sub26nico** [comment]
I have bugs with the mouse: I can't launch/stop/play with left click on a clip., and right click doesn't 
work too. Same with the Scene buttons. I can launch/stop/play clips with my midi controller (Korg NanoPAD2) but not cleaning the clips. The faders and knobs works well with the mouse, the master section and the metronome are fine too.

File: src/avtk/clipselector.cxx [hunk 1]
```diff
@@ -247,10 +334,12 @@ int ClipSelector::handle(int event)
 		{
 			// calculate the clicked clip number
 			int clipHeight = (h / numClips);
-			int clipNum = ( (Fl::event_y() ) - y ) / clipHeight;
+			clipNum = ( (Fl::event_y() ) - y ) / clipHeight;
 			if (clipNum >= numClips)
 				clipNum = numClips -1; // fix for clicking the lowest pixel
 
+			return clipNum;
```
Comment (georgkrause) [inline src/avtk/clipselector.cxx]: this return prevents that mouse events can be handled. why do we need this?

**Comment by vale981** [comment]
I suppose I forgot that return while refactoring.

File: src/avtk/clipselector.cxx (hunk 1 above)
Comment (georgkrause) [reply src/avtk/clipselector.cxx]: I fixed this in my testing branch: https://github.com/georgkrause/openAV-Luppp/commit/9516b79b7b10c0ca6d3feac3d88326d30cf46e9f

seems to work, did no deeper testing.

Comment (sub26nico) [reply src/avtk/clipselector.cxx]: same here, no deep tests but the mouse can launch/stop clips now

**Comment by georgkrause** [comment]
@vale981 i tried to ping you on irc several times, maybe you can ping me when you have time.

**Pull request closed without merging**